pkg/exchange/bitget/stream_test.go

```diff
@@ -0,0 +1,515 @@
+	t.Run("op subscribe event", func(t *testing.T) {
+		input := `{
+		   "event":"subscribe",
+		   "arg":{
+			  "instType":"sp",
+			  "channel":"books5",
+			  "instId":"BTCUSDT"
+		   }
+		}`
+		res, err := parseWebSocketEvent([]byte(input))
+		assert.NoError(t, err)
+		opEvent, ok := res.(*WsEvent)
+		assert.True(t, ok)
+		assert.Equal(t, WsEvent{
+			Event: WsEventSubscribe,
+			Arg: WsArg{
+				InstType: instSp,
+				Channel:  ChannelOrderBook5,
+				InstId:   "BTCUSDT",
+			},
+		}, *opEvent)
+
+		assert.NoError(t, opEvent.IsValid())
+	})
+
+	t.Run("op unsubscribe event", func(t *testing.T) {
+		input := `{
+		   "event":"unsubscribe",
+		   "arg":{
+			  "instType":"sp",
+			  "channel":"books5",
+			  "instId":"BTCUSDT"
+		   }
+		}`
+		res, err := parseWebSocketEvent([]byte(input))
+		assert.NoError(t, err)
+		opEvent, ok := res.(*WsEvent)
+		assert.True(t, ok)
+		assert.Equal(t, WsEvent{
+			Event: WsEventUnsubscribe,
+			Arg: WsArg{
+				InstType: instSp,
+				Channel:  ChannelOrderBook5,
+				InstId:   "BTCUSDT",
+			},
+		}, *opEvent)
+	})
+
+	t.Run("op error event", func(t *testing.T) {
+		input := `{
+		   "event":"error",
+		   "arg":{
+			  "instType":"sp",
+			  "channel":"books5",
+			  "instId":"BTCUSDT-"
+		   },
+		   "code":30001,
+		   "msg":"instType:sp,channel:books5,instId:BTCUSDT- doesn't exist",
+		   "op":"subscribe"
+		}`
+		res, err := parseWebSocketEvent([]byte(input))
+		assert.NoError(t, err)
+		opEvent, ok := res.(*WsEvent)
+		assert.True(t, ok)
+		assert.Equal(t, WsEvent{
+			Event: WsEventError,
+			Code:  30001,
+			Msg:   "instType:sp,channel:books5,instId:BTCUSDT- doesn't exist",
+			Op:    "subscribe",
+			Arg: WsArg{
+				InstType: instSp,
+				Channel:  ChannelOrderBook5,
+				InstId:   "BTCUSDT-",
+			},
+		}, *opEvent)
+	})
+
+	t.Run("Orderbook event", func(t *testing.T) {
+		input := `{
+		   "action":"%s",
+		   "arg":{
+			  "instType":"sp",
+			  "channel":"books5",
+			  "instId":"BTCUSDT"
+		   },
+		   "data":[
+			  {
+				 "asks":[
+					[
+					   "28350.78",
+					   "0.2082"
+					],
+					[
+					   "28350.80",
+					   "0.2081"
+					]
+				 ],
+				 "bids":[
+					[
+					   "28350.70",
+					   "0.5585"
+					],
+					[
+					   "28350.67",
+					   "6.8175"
+					]
+				 ],
+				 "checksum":0,
+				 "ts":"1697593934630"
+			  }
+		   ],
+		   "ts":1697593934630
+		}`
+
+		eventFn := func(in string, actionType ActionType) {
+			res, err := parseWebSocketEvent([]byte(in))
+			assert.NoError(t, err)
+			book, ok := res.(*BookEvent)
+			assert.True(t, ok)
+			assert.Equal(t, BookEvent{
+				Events: []struct {
+					Asks types.PriceVolumeSlice `json:"asks"`
+					// Order book on buy side, descending order
+					Bids     types.PriceVolumeSlice     `json:"bids"`
+					Ts       types.MillisecondTimestamp `json:"ts"`
+					Checksum int                        `json:"checksum"`
+				}{
+					{
+						Asks: []types.PriceVolume{
+							{
+								Price:  fixedpoint.NewFromFloat(28350.78),
```

Comment:
you can check the number() function, I think you can make a copy here for writing the tests

it helps you to create fixedpoint value object with just number(3200.0) or number("3200.0"